### ISSUE TITLE
Too many boxes being proposed

### ISSUE THREAD

**Issue by 17sarf** [pytorch/vision#1309]
I have trained a model using the KAIST dataset. I am using a subset of the full dataset (approximately 2000 samples). I have trained it for over 150 epochs. The loss values etc are quite low:

`Epoch: [156]  [160/515]  eta: 0:13:50  lr: 0.000000  loss: 0.0630 (0.0905)  loss_classifier: 0.0341 (0.0499)  loss_box_reg: 0.0240 (0.0327)  loss_objectness: 0.0007 (0.0018)  loss_rpn_box_reg: 0.0045 (0.0061)  time: 2.2263  data: 0.0173  max mem: 7979`

But when I run an image, such as the one below:

![download](https://user-images.githubusercontent.com/44234385/64479481-873e4d80-d1af-11e9-918c-ac2e7a977718.png)

This the output:

![download (1)](https://user-images.githubusercontent.com/44234385/64479524-41ce5000-d1b0-11e9-98fe-7c4f8c4700c9.png)

I am unsure is this may be part of the issue [#1308](url).

**Comment by fmassa** [comment]
Your model is not learning properly. It might be due to several reasons:
- you are not starting from a pre-trained backbone
- 2000 images are not enough if you don't do transfer learning

I'd recommend you start with the tutorial in https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html , without modifying the model, and just applying it to your own dataset. This will give you some understanding on how the model could potentially work, and will help you modify the model if needed for your application

**Comment by 17sarf** [comment]
Thank you for you reply. I have done as you suggested and I am getting much better results.
